app/src/main/java/com/futurice/freesound/viewmodel/DisplayableItem.java

```diff
@@ -0,0 +1,46 @@
+package com.futurice.freesound.viewmodel;
+
+import com.google.auto.value.AutoValue;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Wraps a model. Convenient for complex presentation layers such as
+ * {@link android.support.v7.widget.RecyclerView} where different types of model are handled.
+ */
+@AutoValue
+public abstract class DisplayableItem<T> {
+
+    // Types
+    public static final int SOUND = 0;
+
+    public abstract int type();
+
+    @NonNull
+    public abstract T model();
+
+    @SuppressWarnings("NullableProblems")
+    @AutoValue.Builder
+    public interface Builder<T> {
+
+        @NonNull
+        Builder<T> type(int type);
+
+        @NonNull
+        Builder<T> model(@NonNull T model);
+
+        @NonNull
+        DisplayableItem<T> build();
+    }
+
+    @NonNull
+    public static <T> Builder<T> builder() {
+        return new AutoValue_DisplayableItem.Builder<>();
+    }
+
+    @NonNull
+    public static DisplayableItem create(@NonNull final Object model,
```

Comment:
This was intended. The methods in the adapter are for plain DisplayableItems, like update(DisplayableItem items) instead of update(DisplayableItem<T> items). If I remember correctly it does not compile if you try to do update(DisplayableItem.<T>create(...)).